### PR TITLE
Worked on https://github.com/drupalcommerce/commerce-docs/issues/106 …

### PR DESCRIPTION
**Pull request by saurabh-vijayvargiya**
Documented process for creating new product type

### CODE REVIEW

File: v2/product/product-type.rst [hunk 1]
```diff
@@ -1,4 +1,68 @@
 Create a product type
```
Comment (joshmiller83) [inline v2/product/product-type.rst]: We need an overview for this section. Give us 2-3 sentence summary of what we're accomplishing and why someone would want to create product types.

File: v2/product/product-type.rst [hunk 2]
```diff
@@ -1,4 +1,68 @@
...
 =====================
+For this example we will create a product type for T-shirts
+## First create Product attribute
```
Comment (joshmiller83) [inline v2/product/product-type.rst]: A good heading here would be "Product Attributes"

Be sure to link to the Product Attributes section for more information. Let the person reading this documentation know why you've chosen to create attributes. They will want to understand your decision so they can use that to make decisions for their own data model. 

For example: 

> Not all products have attributes. For our example product type, we're creating shirts. Shirts usually come in colors and sizes. One type of shirt may have a few sizes that it comes in. Those sizes are what we call attributes. And those attributes can be created like this:

File: v2/product/product-type.rst [hunk 3]
```diff
@@ -1,4 +1,68 @@
...
+First we will need to create a product attribute. We will create two product attributes - 
 
-@todo
+* Size
+* Color
+
+Goto ``admin/commerce/product-attributes``
```
Comment (joshmiller83) [inline v2/product/product-type.rst]: For important menu transitions, you should explain, especially to customers, where they can find this in the menu system. They don't know what URLs are.

File: v2/product/product-type.rst [hunk 4]
```diff
@@ -1,4 +1,68 @@
...
+![](../images/product-attribute-3.png) 
+
+Now save this form.
+
+Do the same for Color attribute.
+
+## Create Product variation
+Now we need a product variation for the image for t-shirts.
+
+Goto ``admin/commerce/config/product-variation-types``
```
Comment (joshmiller83) [inline v2/product/product-type.rst]: Again, it is helpful to help them find this in the menu system, help them click through this whole thing.

File: v2/product/product-type.rst [hunk 5]
```diff
@@ -1,4 +1,68 @@
...
+
+Click **Add product variation type**
+![](../images/product-variation-1.png) 
+
+This will open a form.
+![](../images/product-variation-2.png) 
+
+Now we will add image field to our product variation.
```
Comment (joshmiller83) [inline v2/product/product-type.rst]: "Now we will add ***an*** image field..."

Discuss **why** you are choosing to put the image field on the product variation and not the product. In fact, call that out here in a little aside, give a good reason you would want to choose one over the other, depending on circumstances.

File: v2/product/product-type.rst [hunk 6]
```diff
@@ -1,4 +1,68 @@
...
+Click on **Manage fields** and then click on **Add field**
+![](../images/product-variation-3.png)
+
+Now select field type Image under *Reference*.
+![](../images/product-variation-4.png) 
+
+Then click on **Save and continue**. Then save the settings for image field.
+
+## Create Product Type
+Now lets create our product type T-shirt.
```
Comment (joshmiller83) [inline v2/product/product-type.rst]: We need a little more meat here. What is a product type? Should you create a shirt or tshirt or a clothing product type? Explain to the store administrator how to make a good decision on determining what a product type is.

File: v2/product/product-type.rst [hunk 7]
```diff
@@ -1,4 +1,68 @@
...
+Click on **Add product type**
+![](../images/product-type-1.png) 
+
+This will open a form.
+![](../images/product-type-2.png) 
+
+Save settings and this will create our T-shirt product type.
+
+### Create a test product
+Goto ``admin/commerce/products``. Click **Add product**. Click product type *T-shirt*
```
Comment (joshmiller83) [inline v2/product/product-type.rst]: How do I find this URL using the menu system?

Now would be a good time to tell the store administrator that they should see all of their product types listed here on this screen.

**Comment by saurabh-vijayvargiya** [comment]
@joshmiller83 Pull request has been updated. Please review now.

Thanks

File: v2/product/product-type.rst [hunk 8]
```diff
@@ -1,4 +1,113 @@
 Create a product type
 =====================
+Imagine a new product arrives in your store, let's say T-shirt. You have never sold t-shirt in your store and now you have to sell it in your store. So for this new product, you will need to create a new product type in your drupal commerce site. This is not a hard process, below is a step by step guide.
 
-@todo
+## Product attribute
```
Comment (mglaman) [inline v2/product/product-type.rst]: Headers in ReST are underlined, equally, by `---------`

File: v2/product/product-type.rst [hunk 9]
```diff
@@ -1,4 +1,113 @@
...
+We will create two product attributes - 
+
+* Size
+* Color
+
+Goto ``admin/commerce/product-attributes``
+
+OR
+
+![Click on Product attribute](images/product-attribute-4.png)
```
Comment (mglaman) [inline v2/product/product-type.rst]: `.. figure:` syntax

File: v2/product/product-type.rst [hunk 10]
```diff
@@ -1,4 +1,113 @@
...
+
+Goto ``admin/commerce/product-attributes``
+
+OR
+
+![Click on Product attribute](images/product-attribute-4.png) 
+
+Click **Add product attribute**
+
+![Click on Add product attribute](images/product-attribute-1.png)
```
Comment (mglaman) [inline v2/product/product-type.rst]: `.. figure:` syntax

File: v2/product/product-type.rst [hunk 11]
```diff
@@ -1,4 +1,113 @@
...
+
+![Click on Product attribute](images/product-attribute-4.png) 
+
+Click **Add product attribute**
+
+![Click on Add product attribute](images/product-attribute-1.png) 
+
+This will open a form.
+
+![Fill Product attribute form](images/product-attribute-2.png)
```
Comment (mglaman) [inline v2/product/product-type.rst]: `.. figure:` syntax

File: v2/product/product-type.rst [hunk 12]
```diff
@@ -1,4 +1,113 @@
...
+
+![Click on Add product attribute](images/product-attribute-1.png) 
+
+This will open a form.
+
+![Fill Product attribute form](images/product-attribute-2.png) 
+
+After saving this form, we are required to fill *attribute values*.
+
+![Fill attribute values](images/product-attribute-3.png)
```
Comment (mglaman) [inline v2/product/product-type.rst]: `.. figure:` syntax

File: v2/product/product-type.rst [hunk 13]
```diff
@@ -1,4 +1,113 @@
...
+
+After saving this form, we are required to fill *attribute values*.
+
+![Fill attribute values](images/product-attribute-3.png) 
+
+Now save this form.
+
+Do the same for Color attribute.
+
+## Create Product variation
```
Comment (mglaman) [inline v2/product/product-type.rst]: `-------`

File: v2/product/product-type.rst [hunk 14]
```diff
@@ -1,4 +1,113 @@
...
+Now we need a product variation for the image for t-shirts.
+
+Goto ``admin/commerce/config/product-variation-types``
+
+OR
+
+![Click on Product variation type](images/product-variation-5.png)
```
Comment (mglaman) [inline v2/product/product-type.rst]: `.. figure:` syntax

File: v2/product/product-type.rst [hunk 15]
```diff
@@ -1,4 +1,113 @@
...
+
+Goto ``admin/commerce/config/product-variation-types``
+
+OR
+
+![Click on Product variation type](images/product-variation-5.png) 
+
+Click **Add product variation type**
+
+![Click on Add product variation type](images/product-variation-1.png)
```
Comment (mglaman) [inline v2/product/product-type.rst]: `.. figure:` syntax

**Comment by saurabh-vijayvargiya** [comment]
@joshmiller83 @mglaman Pull request updated.

**Comment by mglaman** [comment]
@saurabh-vijayvargiya I don't see the push

**Comment by saurabh-vijayvargiya** [comment]
@mglaman Sorry, I thought my push would automatically update the previous P.R. Here's new pull request url -> https://github.com/drupalcommerce/commerce-docs/pull/123

**Pull request closed without merging**